### ISSUE TITLE
Add more sanity checks in get_thread_heap_raw

### ISSUE THREAD

**Issue by strlght** [mjansson/rpmalloc#147]
My use case is quite specific – basically I needed to change allocator to rpmalloc in some mach-o object files while keeping default allocator for others (so it's not possible to use `ENABLE_OVERRIDE` for that).
While doing that I encountered that it's possible to add more sanity checks in `get_thread_heap_raw()`: eg. check if `_memory_thread_heap  == 0` before calling `pthread_getspecific(_memory_thread_heap)`.

1. What do you think about that?
2. Any chance that you might have suggestions on how I should approach my problem? Right now I've modified `get_thread_heap_raw()` with this check that I mentioned:
```
static inline heap_t*
get_thread_heap_raw(void) {
#if (defined(__APPLE__) || defined(__HAIKU__)) && ENABLE_PRELOAD
	if (EXPECTED(_memory_thread_heap != 0)) {
		return pthread_getspecific(_memory_thread_heap);
	} else {
		return 0;
	}
#else
	return _memory_thread_heap;
#endif
}
```
It looks like it might the only option to get the same behaviour across all platforms.

**Comment by mjansson** [comment]
Why would that be needed? You need to call rpmalloc_initialize/rpmalloc_initialize_config before anything else, and that will setup the thread local data

**Comment by strlght** [comment]
Let me rephrase that: right now behavior of `get_thread_heap_raw()` is inconsistent across platforms with `ENABLE_PRELOAD=1`.
On Haiku/Apple platforms it will return some value because of undefined behavior in `pthread_getspecific` call. If the returned value is non-zero, then none of the heap initialization code in `get_thread_heap` will be executed.
On every other platform `_memory_thread_heap` is always zero before the call to `rpmalloc_initialize/rpmalloc_initialize_config`. So the heap is initialized in the first call to `get_thread_heap`. 

Is it supposed to be like this?

**Comment by mjansson** [comment]
Yes, calling any other rpmalloc functions before rpmalloc_initialize/rpmalloc_initialize_config is not allowed and will result in undefined behaviour, to avoid polluting all paths with extra checks.

I will make this a bit more clear in the documentation.
